docs/troubleshooting.md

```diff
@@ -0,0 +1,63 @@
+# Troubleshooting handbook
+
+## Introduction
+
+This page will give you some guild on troubleshooting for some situations when you deploy and use the kuberay.
+
+## Ray Version Compatibility
+
+### Problem
+
+For every running ray cluster, when we try to connect with the client, we must be careful about the python and ray version we used. There are several  issues report failures related to version imcompatibility: [#148](https://github.com/ray-project/kuberay/issues/148), [#21549](https://github.com/ray-project/ray/issues/21549). Therefore there is a reminder for troubleshooting when come up with that situation.
+
+### Error cases
+
+In the ray client initialization, there are several checks will be executed in [ray-project/ray/util/client/\_\_init\_\_.py:L115-L137](https://github.com/ray-project/ray/blob/master/python/ray/util/client/__init__.py#L115-L137).
+
+Common cases would be like:
+
+```bash
+...
+
+RuntimeError: Python minor versions differ between client and server: client is 3.8.10, server is 3.7.7
+```
+
+or:
+
+```bash
+...
+RuntimeError: Client Ray installation incompatible with server: client is 2021-05-20, server is 2021-12-07
+```
+
+Some cases may not be so clear:
+
+```bash
+ConnectionAbortedError: Initialization failure from server:
+Traceback (most recent call last):
+...
+'AttributeError: ''JobConfig'' object has no attribute ''_parsed_runtime_env''
+```
+
+```
+Traceback (most recent call last):
+...
+RuntimeError: Version mismatch: The cluster was started with:
```

Comment:
There's another case that even you use same ray version. if the python minor version doesn't match, it would fail as well. I will merge this one as "add trobleshooting for `ray` version mismatch" and please give it a try when you get a chance.

Reply:
@scarlet25151